File: app/src/main/kt/estyle/teabaike/activity/SplashActivity.java
package estyle.teabaike.activity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;

import javax.inject.Inject;

import estyle.teabaike.application.TeaBaikeApplication;
import estyle.teabaike.manager.TimerManager;
import io.reactivex.disposables.Disposable;

public class SplashActivity extends BaseActivity {

    @Inject
    TimerManager mTimerManager;

    private Disposable mDisposable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TeaBaikeApplication.getInstance().getTeaBaikeComponent().inject(this);

        mDisposable = mTimerManager.splash()
                .subscribe(isFirstLogin -> {
                    if (isFirstLogin) {
                        WelcomeActivity.startActivity(SplashActivity.this);
                        SharedPreferences sharedPreferences = getSharedPreferences(TimerManager.CONFIG_NAME,
                                Context.MODE_PRIVATE);
                        sharedPreferences.edit()
                                .putBoolean("is_first_login", false)
                                .apply();
                    } else {
                        MainActivity.startActivity(SplashActivity.this);
                    }
                    finish();
                });
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mDisposable.dispose();
    }
}

File: app/src/main/kt/estyle/teabaike/activity/SplashActivity.kt
package estyle.teabaike.activity

import android.os.Bundle
import androidx.lifecycle.ViewModelProvider
import estyle.base.BaseActivity
import estyle.base.rxjava.DisposableConverter
import estyle.teabaike.config.RoutePath
import estyle.teabaike.viewmodel.SplashViewModel

class SplashActivity : BaseActivity() {

    private val viewModel by lazy { ViewModelProvider(this).get(SplashViewModel::class.java) }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)

        viewModel.delay(2)
            .`as`(DisposableConverter.dispose(this))
            .subscribe { isFirstLogin ->
                if (isFirstLogin) {
                    aRouter.build(RoutePath.LAUNCHER_WELCOME).navigation()
                } else {
                    aRouter.build(RoutePath.LAUNCHER_MAIN).navigation()
                }
                finish()
            }
    }
}